Sources/Honeycomb/HoneycombDeterministicSampler.swift

```diff
@@ -0,0 +1,60 @@
+import Foundation
+import OpenTelemetryApi
+import OpenTelemetrySdk
+
+class HoneycombDeterministicSampler: Sampler {
+    private let inner: Sampler
+    private let rate: [String: AttributeValue]
+
+    init(sampleRate: Int) {
+        var inner: Sampler
+
+        switch sampleRate {
+        case Int.min..<1:
+            inner = Samplers.alwaysOff
+        case 1:
+            inner = Samplers.alwaysOn
+        default:
+            inner = Samplers.traceIdRatio(ratio: 1.0 / Double(sampleRate))
```

Comment:
Why are we doubling the sample rathe here for the ratio? Or am I reading the code wrong 🙈

Reply:
Ah nvm I just don't know how swift works!